### PR TITLE
#1014 use document list to display search results

### PR DESCRIPTION
**Pull request by magemello**
**Please check if the PR fulfills these requirements**
```
[ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
[ ] Tests for the changes have been added (for bug fixes / features)
[ ] Docs have been added / updated (for bug fixes / features)
```

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[ x ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] Documentation
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

#1014 

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[ ] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: ng2-components/ng2-alfresco-documentlist/src/components/document-list.component.ts [hunk 1]
```diff
@@ -67,10 +83,15 @@ export class DocumentListComponent implements OnInit, OnChanges, AfterContentIni
...
     pageSize: number = DocumentListComponent.DEFAULT_PAGE_SIZE;
 
+    skipCount: number = 0;
+
+    pagination: Pagination;
+
     @Input()
     set rowFilter(value: RowFilter) {
-        if (this.data) {
+        if (this.data && value && this.currentFolderId) {
```
Comment (DenisVuyka) [inline ng2-components/ng2-alfresco-documentlist/src/components/document-list.component.ts]: Checking for `value` might mean you can never reset row filter by providing `null`.

File: ng2-components/ng2-alfresco-documentlist/src/components/document-list.component.ts [hunk 2]
```diff
@@ -102,25 +133,17 @@ export class DocumentListComponent implements OnInit, OnChanges, AfterContentIni
...
-    constructor(
-        private documentListService: DocumentListService,
-        private ngZone: NgZone,
-        private translateService: AlfrescoTranslationService) {
+    constructor(private documentListService: DocumentListService,
+                private ngZone: NgZone,
+                private translateService: AlfrescoTranslationService) {
 
-        this.data = new ShareDataTableAdapter(this.documentListService, './', []);
+        this.data = new ShareDataTableAdapter(this.documentListService, this.baseComponentPath, []);
```
Comment (DenisVuyka) [inline ng2-components/ng2-alfresco-documentlist/src/components/document-list.component.ts]: Does it work with SystemJs and Webpack without errors?

File: demo-shell-ng2/app/components/search/search.component.html [hunk 3]
```diff
@@ -1,8 +1,12 @@
 <div class="search-results-container">
     <h1>Search results</h1>
-    <alfresco-search (navigate)="onNavigateItem($event)"></alfresco-search>
+    <alfresco-search (preview)="showFile($event)"></alfresco-search>
```
Comment (DenisVuyka) [inline demo-shell-ng2/app/components/search/search.component.html]: Might be a breaking change.

File: demo-shell-ng2/app/components/search/search.component.ts [hunk 4]
```diff
@@ -47,18 +47,18 @@ import { MinimalNodeEntity } from 'alfresco-js-api';
 })
 export class SearchComponent {
 
+    fileNodeId: any;
```
Comment (DenisVuyka) [inline demo-shell-ng2/app/components/search/search.component.ts]: It it possible putting a type here? I suspect `string` might work.

File: ng2-components/ng2-alfresco-documentlist/src/components/document-list.component.ts (hunk 2 above)
Comment (magemello) [inline ng2-components/ng2-alfresco-documentlist/src/components/document-list.component.ts]: yes

File: demo-shell-ng2/app/components/search/search.component.html (hunk 3 above)
Comment (magemello) [inline demo-shell-ng2/app/components/search/search.component.html]: Thanks, we will track it in the changelog

File: ng2-components/ng2-alfresco-documentlist/src/components/document-list.component.ts (hunk 1 above)
Comment (magemello) [inline ng2-components/ng2-alfresco-documentlist/src/components/document-list.component.ts]: it was checked before inside the adapter....

**Pull request merged**